waku/waku_api/rest/store/handlers.nim

```diff
@@ -209,18 +212,19 @@ proc installStoreV1Handler(router: var RestRouter,
     # /store/v1/messages?peerAddr=%2Fip4%2F127.0.0.1%2Ftcp%2F60001%2Fp2p%2F16Uiu2HAmVFXtAfSj4EiR7mL2KvL4EE2wztuQgUSBoj2Jx2KeXFLN\&pubsubTopic=my-waku-topic
 
     # Parse the peer address parameter
-    var parsedPeerAddr = parseUrlPeerAddr(peerAddr.toOpt())
-    if not parsedPeerAddr.isOk():
-      return RestApiResponse.badRequest(parsedPeerAddr.error)
-
-    var peerOpt = none(RemotePeerInfo)
-    if parsedPeerAddr.value.isSome():
-      peerOpt = parsedPeerAddr.value
-    else:
-      # The user didn't specify any store peer address.
-      peerOpt = node.peerManager.selectPeer(WakuStoreCodec)
-      if peerOpt.isNone():
-        return RestApiResponse.preconditionFailed("Missing known store-peer node")
+    let parsedPeerAddr = parseUrlPeerAddr(peerAddr.toOpt()).valueOr:
+      return RestApiResponse.badRequest(error)
+
+    let peerAddr = parsedPeerAddr.valueOr:
+      node.peerManager.selectPeer(WakuStoreCodec).valueOr:
```

Comment:
Correct me, but seems calling peerManager.selectPeer(...codec) is general among api-s.
Could it be somehow part of the default discovery handler?
Due in case we might be able to avoid to have default none handler arguments in the install procs.

I have the impression, that it is ok to give no handler in general.
If that was a direct aim here having with clear reason, forgive me. ;)

Reply:
I guess it's the same everywhere so it could be default.